crates/distribution-types/src/resolved.rs

```diff
@@ -44,7 +56,29 @@ impl ResolvedDist {
 impl ResolvedDistRef<'_> {
     pub fn to_owned(&self) -> ResolvedDist {
         match self {
-            Self::Installable(dist) => ResolvedDist::Installable((*dist).clone()),
+            Self::InstallableRegistrySourceDist { sdist, prioritized } => {
+                // This is okay because we're only here if the prioritized dist
+                // has an sdist, so this always succeeds.
+                let source = prioritized.source_dist().expect("a source distribution");
```

Comment:
What's the difference between `sdist` and `source` here?

Reply:
Looks like no difference, but `wheel` and `built` _do_ differ in the `Self::InstallableRegistryBuiltDist` case.

Reply:
I see, in the next PR they _do_ differ.

Reply:
Yeah it's a bit weird. My sense is that the types could possibly be clearer here if we could find a way to split `PrioritizedDist` into a "builder phase" and "final phase." But I'm not sure.